### PR TITLE
In out main

### PR DESCRIPTION
**Pull request by ischrot**
I would like to merge my InClass, OutClass and the main notebook back into the oop_formulation branch in which we combine our parts before merging everything into main.

### CODE REVIEW

File: src/in_out_class.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        self.nstate_i = None
+
+        # define methods to read
+        self.read_functions = {
+            "expec.t": lambda path: pd.read_csv(path, r"\s+"),
+            "npop.t": lambda path: pd.read_csv(path, r"\s+"),
+            "table.dat": lambda path: np.loadtxt(path, skiprows=1).T,
+            # Numerics:
+            "efield.t": lambda path: np.loadtxt(path, skiprows=1),
+            "nstate_i.t": lambda path: np.loadtxt(path, skiprows=1),
```
Comment (dcebulla) [inline src/in_out_class.py:28]: Why is data_path not included here? Or where will data_path be used?

**Review by dcebulla — CHANGES_REQUESTED** [review]

File: main.ipynb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    "## Run Statistic Analysis"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "statistics = StatisticsClass(data, thres_stat)\n",
```
Comment (kierandidi) [inline main.ipynb:52]: Do we want StatisticsClass to only accept "data" as argument? Based on the requirements I thought we explicitly pass the different data parts to it. If we just want to use data as argument the __init__ method of StatisticsClass would need to be changed.

**Review by kierandidi — CHANGES_REQUESTED** [review]
Need to check interface between InOut Classes and StatisticsClass

File: main.ipynb (hunk 2 above)
Comment (kierandidi) [reply main.ipynb:52]: Changed the StatisticsClass so that it accepts data as argument

File: src/in_out_class.py (hunk 1 above)
Comment (ischrot) [reply src/in_out_class.py:28]: In l. 54 I use data_path for path. Here, path is just the lambda variabe

File: main.ipynb (hunk 2 above)
Comment (ischrot) [reply main.ipynb:52]: thx

File: src/in_out_class.py (hunk 1 above)
Comment (ischrot) [reply src/in_out_class.py:28]: In l. 54 I use data_path for path. Here, path is just the lambda variable

**Pull request merged**